content/ui-patterns/saving.mdx

```diff
@@ -0,0 +1,157 @@
+
+It's strongly recommended to have a single save button for all of the inputs on the page. If you're designing a page with pieces of content that can be separately toggled into an edit mode (such as editing an issue title), only allow one edit mode toggled on at a time so that only one save button is visible. When there is on save button per section of a page or form, users may be confused about whether the save button will save all of the inputs or just the subset of the inputs. This confusion is even more likely for visually impaired users because they can't see if each save button is visually grouped with a set of inputs.
+
+{/* 
+TODO: find an example of this, mock one up (if that's allowed), or exclude this visual example
+
+![🖼 Include a DO example with image to reinforce message]()
+*/}
+
+{/* TODO: reference the case where there's a declaratively saved form on a page with inherently saved inputs (such as user Profile settings) */}
+
+### State
+
+Do not disable or hide a form's save button if the form is invalid or has not been changed. The only time the save button should be disabled is in cases where you need to prevent the user from taking further action while the request completes (such as updating notification preferences). In these cases, disable the save button and the form inputs. While the form is disabled, update the save button's label with the present continuous form of the action's verb (for example, "Save" becomes "Saving...", "Follow" becomes "Following...") to indicate that they shouldn't navigate away from the current page which may cause the request to fail silently in the event of an error. In most cases, these actions should take minimal time, but in cases where it takes longer than normal, this clear indication is invaluable.
+
+![Screenshot of a CTA label that changes from "Update" to "Updating..."](https://user-images.githubusercontent.com/2313998/150868916-ea209d7c-136e-4f6a-a1b0-51d1e54faf46.png)
+
+## Explicit saving
+An explicit save does not apply changes until the user explicitly takes an action to submit the changes.
+
+If you're designing a form, start with an explicit saving pattern. Avoid mixing paradigms on a single page with multiple forms, and never mix paradigms in a single form. For example: if I upload a profile photo in the "Public profile" form, that change should not be saved until I explicitly submit the whole form.
+
+If there was an error saving the data, the user's data should be preserved in the form and they should be given [feedback](#feedback) about the failure.
+
+{/*
+TODO: determine if we should include the following guidance:
+
+If a user tries to navigate away from a page with an unsaved form, we should warn them that their changes will be lost.
+*/}
+
+### Declarative inputs
+
+When designing a form that is saved explicitly, use the following inputs:
+- Text inputs
+- Checkbox groups
+- Radio button groups
+- Dropdown menus
```

Comment:
Yea, we could be more specific here.

Instead of listing all the possible kinds of dropdowns ([ActionMenu](https://primer.style/react/storybook?path=/story/composite-components-actionmenu2--simple-list-story), [SelectPanel](https://primer.style/react/storybook?path=/story/composite-components-selectpanel--multi-select-story), [DropdownMenu](https://primer.style/react/storybook?path=/story/composite-components-dropdownmenu2--simple-dropdown-menu)), this summary could cover everything:
```suggestion
- Browser-native `<select>` dropdown menus
- Custom dropdown menus that allow multiple selections
```